monitor/services/src/main/java/org/datacleaner/monitor/server/components/InputColumnAndMappedPropertyRewriter.java

```diff
@@ -45,58 +47,138 @@ public boolean rewriteInput(TransformerDescriptor<?> transformer, ProcessStatele
+            if(mappedProp != null) {
+                if(input.configuration == null || !input.configuration.getProperties().containsKey(mappedProp.getName())) {
+                    return enrichMappedProperty(inputColumnProp, mappedProp, input);
+                }
             }
         }
 
+        if(enrichMapDataObjectsToInputColumnsByName(inputColProperties, input)) {
+            return true;
+        }
+
         enrichWithSingleInputProperty(inputColumnProp, input);
         return true;
     }
 
+    private boolean enrichMapDataObjectsToInputColumnsByName(List<ConfiguredPropertyDescriptor> inputColProperties, ProcessStatelessInput input) {
+        if(inputColProperties.isEmpty()) {
+            return false;
+        }
+        // check if we can convert input "map" to the input columns
+        if(input.data.isArray()) {
+            for(JsonNode item: input.data) {
+                if(!item.isObject()) {
+                    return false;
+                }
+            }
+            // OK, we have array of objects
+        } else if(input.data.isObject()) {
+            // OK, we have one object
+        } else {
+            return false;
+        }
+
+        JsonNodeFactory json = Serializator.getJacksonObjectMapper().getNodeFactory();
+        if(input.data.isArray()) {
+            int i = 0;
+            for(JsonNode item: input.data) {
+                ((ArrayNode)input.data).set(i++, mapDataObjectToInputColumnsByName(item, inputColProperties));
+            }
+        } else {
+            input.data = json.arrayNode().add(mapDataObjectToInputColumnsByName(input.data, inputColProperties));
+        }
+
+        if(input.configuration == null) {
+            input.configuration = new ComponentConfiguration();
+        }
+
+        int i = 1;
+        for(ConfiguredPropertyDescriptor inputCol: inputColProperties) {
+            String colName = "c" + i++;
+            input.configuration.getProperties().put(inputCol.getName(), json.textNode(colName));
+            input.configuration.getColumns().add(json.textNode(colName));
+        }
+
+        return true;
+    }
+
+    private JsonNode mapDataObjectToInputColumnsByName(JsonNode item, List<ConfiguredPropertyDescriptor> inputColProperties) {
+        JsonNodeFactory json = Serializator.getJacksonObjectMapper().getNodeFactory();
+        ArrayNode values = json.arrayNode();
+        ITER:
+        for(ConfiguredPropertyDescriptor inputColPropDesc: inputColProperties) {
+            JsonNode value = item.get(inputColPropDesc.getName());
+            if(value != null) {
+                values.add(value);
+                continue;
+            }
+            String[] aliases = inputColPropDesc.getAliases();
+            if(aliases != null) {
+                for (String alias : aliases) {
+                    value = item.get(alias);
+                    if(value != null) {
+                        values.add(value);
+                        continue ITER;
```

Comment:
I'm sorry, but that's going to be a "nope", unless you have a very good reason to used a labeled continue.

Why not just have the `values.add(json.nullNode());` in an else and use break? Might also be a sign that something like a findAlias method could be a good idea.